python/ray/serve/handle.py

```diff
@@ -1,9 +1,37 @@
-from typing import Optional, Dict, Any, Union
+import asyncio
+import threading
+from typing import Coroutine, Optional, Dict, Any, Union
+from contextlib import contextmanager
 
 from ray.serve.context import TaskContext
-from ray.serve.router import RequestMetadata
+from ray.serve.router import RequestMetadata, Router
 from ray.serve.utils import get_random_letters
 
+global_async_loop = None
+
+
+@contextmanager
+def reset_event_loop_policy():
+    policy = asyncio.get_event_loop_policy()
+    asyncio.set_event_loop_policy(None)
+    yield
+    asyncio.set_event_loop_policy(policy)
+
+
+def create_or_get_async_loop_in_thread():
+    global global_async_loop
+    if global_async_loop is None:
+        # We have to asyncio's loop because uvloop will segfault when there
```

Comment:
forgot a word

Reply:
this seems pretty hacky/dangerous. seems like a red flag that we're doing something wrong to me?